mesheryctl/internal/cli/root/app/import.go

```diff
@@ -0,0 +1,211 @@
+	var req *http.Request
+	var app *models.MesheryApplication
+
+	client := &http.Client{}
+
+	// Check if the app manifest is file or URL
+	if validURL := govalidator.IsURL(file); !validURL {
+		content, err := os.ReadFile(file)
+		if err != nil {
+			return nil, err
+		}
+		text := string(content)
+
+		jsonValues, err := json.Marshal(map[string]interface{}{
+			"application_data": map[string]interface{}{
+				"name":             path.Base(file),
+				"application_file": text,
+			},
+			"save": save,
+		})
+		if err != nil {
+			return nil, err
+		}
+		req, err = utils.NewRequest("POST", appURL+"/"+sourceType, bytes.NewBuffer(jsonValues))
+		if err != nil {
+			return nil, err
+		}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		utils.Log.Debug("app file saved")
+		var response []*models.MesheryApplication
+		// failsafe (bad api call)
+		if resp.StatusCode != 200 {
+			return nil, errors.Errorf("Response Status Code %d, possible Server Error", resp.StatusCode)
+		}
+		defer resp.Body.Close()
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Wrap(err, utils.AppError("failed to read response body"))
+		}
+		err = json.Unmarshal(body, &response)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to unmarshal response body")
+		}
+		// set app
+		app = response[0]
+
+	} else {
+		var jsonValues []byte
+		url, path, err := utils.ParseURLGithub(file)
+		if err != nil {
+			return nil, err
+		}
+
+		utils.Log.Debug(url)
+		utils.Log.Debug(path)
+
+		// save the app with Github URL
+		if path != "" {
+			jsonValues, _ = json.Marshal(map[string]interface{}{
+				"url":  url,
+				"path": path,
+				"save": save,
+			})
+		} else {
+			jsonValues, _ = json.Marshal(map[string]interface{}{
+				"url":  url,
+				"save": save,
+			})
+		}
+
+		req, err = utils.NewRequest("POST", appURL+"/"+sourceType, bytes.NewBuffer(jsonValues))
+		if err != nil {
+			return nil, err
+		}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		utils.Log.Debug("remote hosted app request success")
+		var response []*models.MesheryApplication
+		// failsafe (bad api call)
+		if resp.StatusCode != 200 {
+			return nil, errors.Errorf("Response Status Code %d, possible Server Error", resp.StatusCode)
+		}
+		defer resp.Body.Close()
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Wrap(err, utils.AppError("failed to read response body"))
+		}
+		err = json.Unmarshal(body, &response)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to unmarshal response body")
+		}
+
+		// set app
+		app = response[0]
+	}
+
+	return app, nil
+}
+
+func init() {
+	importCmd.Flags().StringVarP(&file, "file", "f", "", "Path/URL to app file")
+	importCmd.Flags().StringVarP(&sourceType, "source-type", "s", "", "Type of source file (ex. manifest / compose / helm)")
```

Comment:
The variables `file` and `sourceType` are not declared in this file but are in other commands. `sourceType` is declared in `onboard.go` and `app.go`. Shouldn't these be changed to variables that are declared in this file itself? 

PS: I'm new to golang & am unaware of the coding styles we follow. So, I'm not really sure of this.

Reply:
The reason they aren't declared is because they are already referenced in `onboard.go` in this struct

https://github.com/meshery/meshery/blob/55dd06a86605980ccb7be3a86a4dd1bb3903d0ac/mesheryctl/internal/cli/root/app/onboard.go#L24-L28

So, they don't need to be declared again :)

Reply:
And even in this struct on `app.go`

https://github.com/meshery/meshery/blob/55dd06a86605980ccb7be3a86a4dd1bb3903d0ac/mesheryctl/internal/cli/root/app/app.go#L11-L14

Reply:
Yup, I'm aware of that. That's why I have this doubt, is assigning a value to a variable for other sibling subcommand fine?

Reply:
It's not an issue 👍 since they are within the same package, reusing them won't cause trouble except if you try to re-declare them :)

Reply:
@nishantwrp, Just like @alphaX86 said, since both the commands are in the same package we can use the variables in both.
In Go, any names declared in a package are accessible in the entire package